File: Esercizi/typescript/esercitazionets/Persona.ts
interface Persona {
    nome: string;
    cognome: string;
    indirizzo: string;
    eta: number;

    getDati(): string;
}

export class Alunno implements Persona {
    constructor(
       public nome: string,
       public cognome: string,
       public eta: number,) { }
    indirizzo: string;
    getDati(): string {
        return `nome: ${this.nome}, cognome: ${this.cognome}, eta: ${this.eta},`
    }

}

File: Java/EsercitazioneString/src/Persona.java
public class Persona {
    String nome;
    String cognome;
    int eta;

    public Persona() {
    }

    public Persona(String nome, String cognome, int eta) {
        this.nome = nome;
        this.cognome = cognome;
        this.eta = eta;
    }


    public boolean verificaMaggiorenne() {
        return this.eta >= 18;
    }


    public boolean verificaMaggiorenne(int limite) {
        return this.eta >= limite;
    }

    public static void sommaNumeri(int x, int y) {
        int somma = x + y;
        System.out.println(somma);
    }

    public static void sommaNumeri(int x, int y, int z) {
        int somma = x + y + z;
        System.out.println(somma);
    }

    public static void sommaNumeri(double x, double y) {
        double somma = x + y;
        System.out.println(somma);
    }


}
